### ISSUE TITLE
Create TSLint rule to prevent usage of params for animations 

### ISSUE THREAD

**Issue by josephperrott** [angular/components#15306]
Create TSLint rule to prevent usage of params for animations.  The intention is for this lint rule to be in place to require us to explicitly state why we want to use the param and ensure we declare the accompanying `const` for the parameter being used.

**Comment by devversion** [comment]
Does that mean we want a lint rule that always prevents the usage of params for animations.. and if we want to use them, we need to explicitly disable the lint rule?

**Comment by josephperrott** [comment]
Yes, it is needed because we don't have a way for us to inform closure compiler internally that the property should not be obfuscated.

**Comment by crisbeto** [comment]
No longer relevant.
